### PR TITLE
Fix hydra runner

### PR DESCRIPTION
**Pull request by tkornuta-nvidia**
This PR fixes hydra runner, enabling to use schemas with manual passing of configs files using command line.

- [x] Modified hydra_runner so it injects schema into the right "place" in ConfigStore.
- [x] Covers runner with integration tests

Usage: please take a look at tests and the used "example"  (`tests/hydra/my_app.py`)

Additionally cleans up some hydra settings in NeMo:
- [x] Sets hydra logging outputs only to stdout (no empty log file in  . dir!!)
- [x] Disables the creation of .hydra subdir 
- [x] Removes the old set_config wrapper

### CODE REVIEW

File: tests/hydra/test_hydra_runner.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+# limitations under the License.
+
+import subprocess
+import sys
+from os import path
+
+import pytest
+
+
+class TestHydraRunner:
```
Comment (blisc) [inline tests/hydra/test_hydra_runner.py:22]: Can you add a check that checks if no `.hydra` folders are created?

File: nemo/core/config/hydra_runner.py [hunk 2]
```diff
@@ -51,14 +46,42 @@ def wrapper(cfg_passthrough: Optional[DictConfig] = None) -> Any:
...
                 # Get overriding args in dot string format
                 overrides = parsed_args.overrides  # type: list
 
-                # Update overrides
+                # Disable the creation of .hydra subdir
+                # https://hydra.cc/docs/tutorials/basic/running_your_app/working_directory
+                overrides.append("hydra.output_subdir=null")
+                # Hydra logging outputs only to stdout (no log file).
+                # https://hydra.cc/docs/configure_hydra/logging
+                overrides.append("hydra/job_logging=stdout")
```
Comment (blisc) [inline nemo/core/config/hydra_runner.py:59]: So long as this stops hydra from interfering with the root logger, I am happy with this

Comment (blisc) [reply nemo/core/config/hydra_runner.py:59]: Resolved after discussion

Comment (titu1994) [inline nemo/core/config/hydra_runner.py:59]: Is this supposed to be hydra.job_logging or hydra/job_logging?

Comment (tkornuta-nvidia) [reply nemo/core/config/hydra_runner.py:59]: "hydra/job_logging" -> please take a look at the link I have added above the line ;)

File: tests/hydra/test_hydra_runner.py (hunk 1 above)
Comment (tkornuta-nvidia) [reply tests/hydra/test_hydra_runner.py:22]: good idea, will do

Comment (tkornuta-nvidia) [reply tests/hydra/test_hydra_runner.py:22]: Refined two tests, so they check the absence of both `.hydra` dir default log file.
When reverted one or the other change, the tests fail:
```
tests/hydra/test_hydra_runner.py::TestHydraRunner::test_no_config PASSED                                                                                                                             [ 16%]
tests/hydra/test_hydra_runner.py::TestHydraRunner::test_config1 FAILED                                                                                                                               [ 33%]
tests/hydra/test_hydra_runner.py::TestHydraRunner::test_config1_invalid PASSED                                                                                                                       [ 50%]
tests/hydra/test_hydra_runner.py::TestHydraRunner::test_config2 FAILED                                                                                                                               [ 66%]
tests/hydra/test_hydra_runner.py::TestHydraRunner::test_config2_invalid PASSED                                                                                                                       [ 83%]
tests/hydra/test_hydra_runner.py::TestHydraRunner::test_config2_filepath_schema PASSED                                                                                                               [100%]
```

**Review by blisc — APPROVED** [review]

**Pull request merged**